doc/charm++/manual.rst

```diff
@@ -2555,173 +2557,199 @@ Examples are in ``examples/charm++/load_balancing`` and
+neighboring processors and there is no messaging outside this local
+neighborhood. However, they will not, in general, provide an immediate
+restoration for load balance - the process is iterated until the
+desired load balance can be achieved.
 
 Listed below are the distributed load balancers:
 
--  **NeighborLB**: A neighborhood load balancer in which each processor
-   tries to average out its load only among its neighbors.
-
--  **WSLB**: A load balancer for workstation clusters, which can detect
-   load changes on desktops (and other timeshared processors) and adjust
-   load without interfering with other’s use of the desktop.
-
 -  **DistributedLB**: A load balancer which uses partial information
    about underloaded and overloaded processors in the system to do
    probabilistic transfer of load. This is a refinement based strategy.
 
-An example of a hierarchical strategy can be found in:
-
--  **HybridLB**: This calls GreedyLB at the lower level and RefineLB at
-   the root.
-
-Listed below are load balancers for debugging purposes:
+Custom strategies should be built using TreeLB or DistBaseLB (the base
+class for DistributedLB). Custom strategies that are based on
+CentralLB or HybridBaseLB will continue to be supported for now, but
+support for these will likely be dropped in a future release.
 
--  **RandCentLB**: Randomly assigns objects to processors;
-
--  **RotateLB**: This strategy moves objects to the next available PE
-   every time it is called. It is useful for debugging PUP routines and
-   other migration related bugs.
+All built-in load balancers that do not require external dependencies
+(that is, all of the above load balancers except for ScotchLB) are
+built by default. To use load balancing, users must link a load
+balancing module with their application and pass the appropriate
+runtime flags. See :numref:`lbOption` for details on how to
+use load balancing.
 
 Users can choose any load balancing strategy they think is appropriate
-for their application. We recommend using GreedyRefineLB with
+for their application. We recommend using TreeLB with GreedyRefine for
 applications in general. For applications where the cost of migrating
 objects is very high, say, due to frequent load balancing to handle
-frequent load changes or due to size of data in the object being large,
-a strategy which favors migration minimization at the cost of balance
-(eg: RefineLB) is more suitable. DistributedLB and HybridLB are suitable
-for large number of nodes. Communication-aware load balancers like
-MetisLB and ScotchLB are suitable for communication intensive
-applications. RotateLB and RandCentLB are more useful for debugging
-object migrations. The compiler and runtime options are described in
+frequent load changes or due to size of data in the object being
+large, a strategy which favors migration minimization at the cost of
+balance (eg: RefineLB) is more suitable. DistributedLB is suitable for
```

Comment:
Does Refine currently work? I don't ever remember hearing it was back to a good state. Last I checked it was horribly slow. Unusably so.

Reply:
The new RefineLB implementations should be reasonable, the old CentralLB version was the one with a problem, I think. I'll test it again to make sure.